Lib/enum.py

```diff
@@ -728,6 +728,10 @@ def __contains__(self, other):
                     type(other).__qualname__, self.__class__.__qualname__))
         return other._value_ & self._value_ == other._value_
 
+    def __iter__(self):
+        members, extra_flags = _decompose(self.__class__, self.value)
+        return (m for m in members if m._value_ != 0)
+
```

Comment:
I think the use of `__iter__()` is problematic given that `__contains__()` has different semantics.

`__contains__()` returns true for members with value 0 or compound members, and `__iter__()` does not.  It's bad form for these not to match.

```
>>> class Foo(Flag):
...     NONE = 0
...     A = 1
...     B = 2
...     C = 4
...     C2 = C
...     AB = A | B
>>> x = Foo.A | Foo.B
>>> list(x)
[<Foo.B: 2>, <Foo.A: 1>]
>>> Foo.NONE in x
True
>>> Foo.AB in x
True
```

I'd argue that `__contains__()` is completely broken and no one wants that behavior.  But assuming that can't be changed, using `__iter__()` to iterate the bits is probably not a good idea.

Reply:
I agree that the zero-value flag should not be contained in a non-zero flag, but I have no problem with a compound flag showing up if its bits are present.  You are welcome to try and convince me otherwise, but it will take more than "no one wants that behavior".

Reply:
Hmm, I just learned that `Flag.__contains__()` is technically a subset operator:

```python3
>>> class Foo(Flag):
...     NONE = 0
...     A = 1
...     B = 2
...     C = 4
...     AB = A | B
>>> Foo.AB in (Foo.AB | Foo.C)
True
>>> (Foo.A|Foo.C) in (Foo.AB | Foo.C)
True
```

If that's the case, then actually it's OK for `Foo.None in Foo.A` to be True, since `Foo.None` is the empty set:
```python3
>>> set().issubset({1, 2})
True
```

I would not have made the API this way.  `__contains__` should be strictly for testing whether a bit is in the value (a Flag value is a set of enabled bits, in my mind), while `issubset()` and operator `<=` should be used for subset.